### PR TITLE
Remove travis and add circleci and update go to 1.17

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Remove travis and add circleci and update go to 1.17

#### Ticket Link
N/a

### CODE REVIEW

File: targets/syslog_test.go [hunk 1]
```diff
@@ -48,6 +49,7 @@ func ExampleSyslog() {
 }
 
 func TestSyslogPlain(t *testing.T) {
+	t.Skip("Skipping to debug why is failing on CI")
```
Comment (wiggin77) [inline targets/syslog_test.go:52]: Local syslog is probably not available in test container.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   Thanks.

File: targets/syslog_test.go (hunk 1 above)
Comment (cpanato) [reply targets/syslog_test.go:52]: humm, good to know, i will check that! thanks for the insight and for the review

Comment (cpanato) [reply targets/syslog_test.go]: added the test back :)

**Pull request merged**